sampling/chain_randomness_test.go

```diff
@@ -20,80 +21,92 @@ func TestSamplingChainRandomness(t *testing.T) {
-		chain := testhelpers.RequireTipSetChain(t, 20)
+		builder, ch := makeChain(t, 21)
 
 		// Sample height after the head falls back to the head, and then looks back from there
-		r, err := sampling.SampleChainRandomness(types.NewBlockHeight(uint64(25)), chain)
+		r, err := sampling.SampleChainRandomness(types.NewBlockHeight(uint64(25)), ch)
 		assert.NoError(t, err)
 		assert.Equal(t, []byte(strconv.Itoa(17)), r)
 
 		// Add new head so as to produce null blocks between 20 and 25
 		// i.e.: 25 20 19 18 ... 0
-		headAfterNulls := types.NewBlockForTest(chain[0].ToSlice()[0], uint64(0))
-		headAfterNulls.Height = types.Uint64(uint64(25))
-		headAfterNulls.Ticket = []byte(strconv.Itoa(int(headAfterNulls.Height)))
-		chain = append([]types.TipSet{types.RequireNewTipSet(t, headAfterNulls)}, chain...)
+		headAfterNulls := builder.BuildOn(ch[0], func(b *chain.BlockBuilder) {
+			b.IncHeight(4)
+			b.SetTicket(types.Signature(strconv.Itoa(25)))
+		})
+		ch = append([]types.TipSet{headAfterNulls}, ch...)
 
 		// Sampling in the nulls falls back to the last non-null
-		r, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(24)), chain)
+		r, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(24)), ch)
 		assert.NoError(t, err)
 		assert.Equal(t, []byte(strconv.Itoa(17)), r)
 
 		// When sampling immediately after the nulls, the look-back skips the nulls (not counting them).
-		r, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(25)), chain)
+		r, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(25)), ch)
 		assert.NoError(t, err)
 		assert.Equal(t, []byte(strconv.Itoa(18)), r)
 	})
 
 	t.Run("fails when chain insufficient", func(t *testing.T) {
 		// Chain: 20, 19, 18, 17, 16
 		// The final tipset is not of height zero (genesis)
-		chain := testhelpers.RequireTipSetChain(t, 20)[:5]
+		_, ch := makeChain(t, 21)
+		ch = ch[:5]
 
 		// Sample is out of range
-		_, err := sampling.SampleChainRandomness(types.NewBlockHeight(uint64(15)), chain)
+		_, err := sampling.SampleChainRandomness(types.NewBlockHeight(uint64(15)), ch)
 		assert.Error(t, err)
 
 		// Sample minus lookback is out of range
-		_, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(16)), chain)
+		_, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(16)), ch)
 		assert.Error(t, err)
-		_, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(18)), chain)
+		_, err = sampling.SampleChainRandomness(types.NewBlockHeight(uint64(18)), ch)
 		assert.Error(t, err)
 
 		// Ok when the chain is just sufficiently long.
-		r, err := sampling.SampleChainRandomness(types.NewBlockHeight(uint64(19)), chain)
+		r, err := sampling.SampleChainRandomness(types.NewBlockHeight(uint64(19)), ch)
 		assert.NoError(t, err)
 		assert.Equal(t, []byte(strconv.Itoa(16)), r)
 	})
 
 	t.Run("falls back to genesis block", func(t *testing.T) {
-		chain := testhelpers.RequireTipSetChain(t, 5)
+		_, ch := makeChain(t, 6)
```

Comment:
any reason we are increasing this value (5 -> 6)?

Reply:
Yes, `RequireTipSetChain` misleadingly created a chain one longer than that requested, i.e. the genesis plus the requested count.